firmware/application/ui_navigation.cpp

```diff
@@ -554,6 +559,31 @@ void SystemStatusView::rtc_battery_workaround() {
     }
 }
 
+void SystemStatusView::new_sdcard_structure_checker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path resources_dir = u"RES";
+    std::vector<std::filesystem::path> directories = scan_root_directories(root_dir);
+
+    auto scan_result = std::find(directories.begin(), directories.end(), resources_dir);
+
+    auto sd_status = sd_card::status();
+
+    if (
+        nav_.is_valid() &&
+        (scan_result == directories.end()) &&
+        !(sd_status == Status::NotPresent || sd_status == Status::Present)) {
```

Comment:
I recommend changing this line to:
`(sd_card::status() == sd_card::Status::Mounted)`

Otherwise you will still be displaying the message when the SD Card has an error.

Reply:
But that why I wanted to implement: sdcard error ain’t meant not inserted.

Reply:
Ohhh I got you, sdcard error && RES not exist will pop it.

Reply:
suggestion taken